backend/src/Service/IncomingMail/IncomingMailService.php

```diff
@@ -98,7 +97,31 @@ public function handleIncomingBounce(
                 $bounceObject = new BounceDto($dsnInput->ReadableText, $recipient->Status);
                 $this->ed->dispatch(new IncomingBounceEvent($send, $sendRecipient, $bounceObject));
             } elseif ($smtpResponseParser->isInfrastructureError()) {
-                // TODO: record infra bounces #291
+                $send = $this->sendService->getSendByUuid($bounceUuid);
```

Comment:
Why repeat the same logic (getting the send and send recipient) in the previous branch?  Try to consolidate them. Early returning if it is not a recipient bounce and an infrastructuer bounce should work better